crates/ty_ide/src/goto.rs

```diff
@@ -828,6 +867,495 @@ f(**kwargs<CURSOR>)
+          |
+        3 |             ab = 2
+        4 |             ab = 3
+        5 |             print(ab)
+          |                   ^^
+          |
+        ");
+    }
+
+    #[test]
+    fn goto_def_local_load_cond_rebind() {
+        let test = cursor_test(
+            r#"
+            ab = 1
+            if cond:
+                ab = 2
+            print(a<CURSOR>b)
+        "#,
+        );
+
+        assert_snapshot!(test.goto_definition(), @r"
+        info[goto-type-definition]: Type definition
+         --> main.py:2:13
+          |
+        2 |             ab = 1
+          |             ^^
+        3 |             if cond:
+        4 |                 ab = 2
+          |
+        info: Source
+         --> main.py:5:19
+          |
+        3 |             if cond:
+        4 |                 ab = 2
+        5 |             print(ab)
+          |                   ^^
+          |
+
+        info[goto-type-definition]: Type definition
+         --> main.py:4:17
+          |
+        2 |             ab = 1
+        3 |             if cond:
+        4 |                 ab = 2
+          |                 ^^
+        5 |             print(ab)
+          |
+        info: Source
+         --> main.py:5:19
+          |
+        3 |             if cond:
+        4 |                 ab = 2
+        5 |             print(ab)
+          |                   ^^
+          |
+        ");
+    }
+
+    #[test]
+    fn goto_def_local_load_exhaustive_bind() {
+        let test = cursor_test(
+            r#"
+            if cond:
+                ab = 2
+            else:
+                ab = 1
+            print(a<CURSOR>b)
+        "#,
+        );
+
+        assert_snapshot!(test.goto_definition(), @r"
+        info[goto-type-definition]: Type definition
+         --> main.py:3:17
+          |
+        2 |             if cond:
+        3 |                 ab = 2
+          |                 ^^
+        4 |             else:
+        5 |                 ab = 1
+          |
+        info: Source
+         --> main.py:6:19
+          |
+        4 |             else:
+        5 |                 ab = 1
+        6 |             print(ab)
+          |                   ^^
+          |
+
+        info[goto-type-definition]: Type definition
+         --> main.py:5:17
+          |
+        3 |                 ab = 2
+        4 |             else:
+        5 |                 ab = 1
+          |                 ^^
+        6 |             print(ab)
+          |
+        info: Source
+         --> main.py:6:19
+          |
+        4 |             else:
+        5 |                 ab = 1
+        6 |             print(ab)
+          |                   ^^
+          |
+        ");
+    }
+
+    #[test]
+    fn goto_def_local_load_only_decl() {
+        let test = cursor_test(
+            r#"
+            ab: int
+            print(a<CURSOR>b)
+        "#,
+        );
+
+        assert_snapshot!(test.goto_definition(), @"No definitions found");
```

Comment:
In this case there's no bindings for `ab` so we fail to yield anything (suboptimal).

Reply:
This would be part of goto declarations instead :)